gbif-occurrence-linkage/main.py

```diff
@@ -2,102 +2,70 @@
+        job_id: str) -> dict:
     """
     Map the result of the API call to an annotation
     :param specimen_data: The JSON value of the Digital Specimen
     :param result: The result which contains either the GBIF ID or an error message
     :param job_id: The job ID of the MAS
     :return: Returns a formatted annotation Record which includes the Job ID
     """
-    timestamp = timestamp_now()
-    oa_value = {
-        'entityRelationships': {
-            'entityRelationshipType': 'hasGbifID',
-            'objectEntityIri': f'https://www.gbif.org/occurrence/{result["gbifId"]}',
-            'entityRelationshipDate': timestamp,
-            'entityRelationshipCreatorName': os.environ.get('MAS_NAME'),
-            'entityRelationshipCreatorId': f"https://hdl.handle.net/{os.environ.get('MAS_ID')}"
-        }
-    }
-    annotation = {
-        'rdf:type': 'Annotation',
-        'oa:motivation': 'ods:adding',
-        'oa:creator': {
-            ODS_TYPE: 'oa:SoftwareAgent',
-            'foaf:name': os.environ.get('MAS_NAME'),
-            ODS_ID: f"https://hdl.handle.net/{os.environ.get('MAS_ID')}"
-        },
-        'dcterms:created': timestamp,
-        'oa:target': {
-            ODS_ID: specimen_data[ODS_ID],
-            ODS_TYPE: specimen_data[ODS_TYPE],
-            'oa:selector': {
-                ODS_TYPE: 'ClassSelector',
-                'oa:class': '$.entityRelationships'
-            },
-        },
-        'oa:body': {
-            ODS_TYPE: 'TextualBody',
-            'oa:value': [json.dumps(oa_value)],
-            'dcterms:reference': result['queryString']
-        }
-    }
-    mas_job_record = {
-        "jobId": job_id,
-        "annotations": [annotation]
+    timestamp = shared.timestamp_now()
+    ods_agent = shared.get_agent()
+    oa_value = shared.map_to_entity_relationship('hasGbifID',
+                                          f'https://www.gbif.org/occurrence/{result["gbifId"]}',
+                                          timestamp, ods_agent)
+    oa_selector = shared.build_class_selector('$.ods:hasEntityRelationship')
+    annotation = shared.map_to_annotation(ods_agent, timestamp, oa_value, oa_selector,
+                                   specimen_data[shared.ODS_ID],
+                                   specimen_data[shared.ODS_TYPE],
+                                   result['queryString'])
+
+    return {
+        'jobId': job_id,
+        'annotations': [annotation]
     }
-    return mas_job_record
-
-
-def timestamp_now() -> str:
-    """
-    Create a timestamp in the correct format
-    :return: The timestamp as a string
-    """
-    timestamp = str(datetime.now(tz=timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.%f"))
-    timestamp_cleaned = timestamp[:-3]
-    timestamp_timezone = timestamp_cleaned + 'Z'
-    return timestamp_timezone
 
 
-def send_updated_opends(annotation: Dict, producer: KafkaProducer) -> None:
+def publish_annotation_event(annotation: Dict, producer: KafkaProducer) -> None:
```

Comment:
Maybe make a type for annotation based on the schema?

Reply:
maybe in another PR, not sure if there exists a jsonschema2PythonType library or something like that so not sure if it would be worth it right now